File: src/utils/kerberosMeta.js
/** IITD student ids: aa1234567 (2 letters + 7 digits) or abc123456 (3 letters + 6 digits). */
export const STUDENT_KERBEROS_RE = /^(?:[a-z]{2}[0-9]{7}|[a-z]{3}[0-9]{6})$/;

export function isStudentKerberos(kerberos) {
    return STUDENT_KERBEROS_RE.test((kerberos || '').toLowerCase().trim());
}

/** Branch prefix + entry year from IITD kerberos id (matches BranchAnalytics). */
export function kerberosMeta(kerberos) {
    const id = (kerberos || '').trim();
    const match = id.match(/^([a-z0-9]{3})([0-9]{2})/i);
    if (!match) {
        return { branch: null, entryYear: null };
    }
    return {
        branch: match[1].toUpperCase(),
        entryYear: `20${match[2]}`,
    };
}

File: app/lib/core/kerberos_meta.dart
class KerberosMeta {
  const KerberosMeta({this.branch, this.entryYear});

  final String? branch;
  final String? entryYear;
}

KerberosMeta kerberosMeta(String kerberos) {
  final match = RegExp(r'^([a-z0-9]{3})([0-9]{2})', caseSensitive: false).firstMatch(kerberos.trim());
  if (match == null) {
    return const KerberosMeta();
  }
  return KerberosMeta(
    branch: match.group(1)!.toUpperCase(),
    entryYear: '20${match.group(2)!}',
  );
}

String formatHostel(String? hostel) {
  final value = hostel?.trim() ?? '';
  return value.isEmpty ? '—' : value;
}
